### ISSUE TITLE
Package-lock out of sync

### ISSUE THREAD

**Issue by mtrezza** [parse-community/Parse-SDK-JS#1515]
### New Issue Checklist

- [x] I am not disclosing a [vulnerability](https://github.com/parse-community/parse-server/blob/master/SECURITY.md).
- [x] I am not just asking a [question](https://github.com/parse-community/.github/blob/master/SUPPORT.md).
- [x] I have searched through [existing issues](https://github.com/parse-community/Parse-SDK-JS/issues?q=is%3Aissue).
- [x] I can reproduce the issue with the latest versions of [Parse Server](https://github.com/parse-community/parse-server/releases) and the [Parse JS SDK](https://github.com/parse-community/Parse-SDK-JS/releases). 

### Issue Description

CI fails in various PRs with:
```
npm ERR! cipm can only install packages when your package.json and package-lock.json or npm-shrinkwrap.json are in sync. Please update your lock file with `npm install` before continuing.
```

### Steps to reproduce
See https://github.com/parse-community/Parse-SDK-JS/runs/7474143947?check_suite_focus=true

### Actual Outcome

Tests fail

### Expected Outcome

Tests pass

###  Environment
Current alpha

n/a

### Logs
n/a

**Comment by mtrezza** [comment]
@parse-community/js-sdk could someone take a look at this? The tests started to fail since recently. Regenerating the package-lock file repeatedly produces the following 3 failing tests:

```
1) Dirty Objects tracks dirty arrays
- Failed: 1 == 2
2) Dirty Objects tracks dirty objects
- Unhandled promise rejection: AssertionError [ERR_ASSERTION]: 1 == 2
3) Dirty Objects tracks dirty geo points
- Unhandled promise rejection: AssertionError [ERR_ASSERTION]: 5 == 10
```

See https://github.com/parse-community/Parse-SDK-JS/runs/7474143947?check_suite_focus=true

This could be an issue in the Parse JS SDK, but it may as well be an issue in the current Parse Server alpha branch, which is the branch that is used to run the Parse JS SDK tests.

**Comment by dblythy** [comment]
I really can't work out why these tests are failing due to a new package-lock.

The failing tests allow Parse Object to set updated objects `without` having to call .save (tracks in memory).

```js
    const geo = new Parse.GeoPoint(5, 5);
    const object = new TestObject();
    object.set('location', geo);
    await object.save()
    geo.latitude = 10;
    // do not have to call .set again
    await object.save();
    const query = new Parse.Query(TestObject);
    const o = await query.get(object.id);
    assert.equal(o.get('location').latitude, 10);
```

Not sure which dependancy would cause this.

Any ideas?

**Comment by mtrezza** [comment]
> I really can't work out why these tests are failing due to a new package-lock.

I don't think the tests are failing due to the regenerated package-lock. It seems that resolving the package-lock issue only uncovered the failing tests.

**Comment by mtrezza** [comment]
You could either debug-step through the tests (do they pass locally and only fail remotely?), to investigate why they fail. Or you could just to back in the commit history to find the last commit where the tests pass, then look at the difference to the next commit.

**Comment by dblythy** [comment]
Is this still an issue @mtrezza?

**Comment by mtrezza** [comment]
Yes, it is. I wasn't able to track this down. To replicate this:
1. fork `alpha`
2. delete `package-lock.json`
3. run `npm i`

The CI should then fail.

That may be related to the fact that synk and dependabot manipulate the package-lock file and we are now in a state where the current (manipulated) file is significant differently than the file when recreated. So we are now going with the current manipulated file and doing releases, but we should be able to recreate package-lock file at any point in time and have the CI pass.

**Comment by dblythy** [comment]
After some tedious investigation, It seems like the issue is caused by Parse Server. CI is currently locked to the latest version, yet `npm i` causes it to fetch `alpha`.

It seems like the commit that causes the failure is [here](https://github.com/parse-community/parse-server/commit/19900fcdf8c9f29a674fb62cf6e4b3341d796891).

The PR that should fix this issue is [here](https://github.com/parse-community/parse-server/pull/8159)

**Comment by dblythy** [comment]
I think this was closed a bit too soon, it cannot be closed until the upstream bug in Parse Server is resolved by https://github.com/parse-community/parse-server/pull/8159. Once https://github.com/parse-community/parse-server/pull/8159 is merged, Dependabot will create a new PR pointing to the updated `alpha` branch, and then the tests should pass. Or, once it is merged, we can regenerate `package-lock` from a fresh `npm i`

**Comment by parseplatformorg** [comment]
🎉 This change has been released in version [3.5.0-alpha.4](https://github.com/parse-community/Parse-SDK-JS/releases/tag/3.5.0-alpha.4)

**Comment by parseplatformorg** [comment]
🎉 This change has been released in version [3.5.0-beta.1](https://github.com/parse-community/Parse-SDK-JS/releases/tag/3.5.0-beta.1)

**Comment by parseplatformorg** [comment]
🎉 This change has been released in version [3.5.0](https://github.com/parse-community/Parse-SDK-JS/releases/tag/3.5.0)
